src/Tests/Cache/ShardedPhpRedisUnitTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @file
+ * Definition of Drupal\redis\Tests\Cache\PhpRedisUnitTest.
+ */
+
+namespace Drupal\redis\Tests\Cache;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Site\Settings;
+use Drupal\redis\Cache\ShardedPhpRedis;
+use Drupal\system\Tests\Cache\GenericCacheBackendUnitTestBase;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Tests PhpRedis cache backend using GenericCacheBackendUnitTestBase.
+ *
+ * @group redis
+ */
+class ShardedPhpRedisUnitTest extends GenericCacheBackendUnitTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = array('system', 'redis');
+
+  public function containerBuild(ContainerBuilder $container) {
+    parent::containerBuild($container);
+    // Replace the default checksum service with the redis implementation.
+    if ($container->has('redis.factory')) {
+      $container->register('cache_tags.invalidator.checksum', 'Drupal\redis\Cache\RedisCacheTagsChecksum')
+        ->addArgument(new Reference('redis.factory'))
+        ->addTag('cache_tags_invalidator');
+    }
+  }
+
+
+  /**
+   * Creates a new instance of PhpRedis cache backend.
+   *
+   * @return \Drupal\redis\Cache\PhpRedis
+   *   A new PhpRedis cache backend.
+   */
+  protected function createCacheBackend($bin) {
+    $cache = new ShardedPhpRedis(
```

Comment:
I was forced to do it this way I couldn't find how to modify Settings class content in order to force it to use my own implementation. I think the other test case for your backend should probably fixed the same way too.